src/resolver.cpp

```diff
@@ -165,7 +267,14 @@ ArAssetInfo UsdOpenAssetIOResolver::_GetAssetInfo(const std::string &assetPath,
 
 ArTimestamp UsdOpenAssetIOResolver::_GetModificationTimestamp(
     const std::string &assetPath, const ArResolvedPath &resolvedPath) const {
-  auto result = ArDefaultResolver::_GetModificationTimestamp(assetPath, resolvedPath);
+  ArTimestamp result;
+  if (manager_->isEntityReferenceString(assetPath)) {
+    // Deliberately use a valid fixed timestamp, to force caching.
+    // TODO(DF): Consider how best to handle this via OpenAssetIO.
```

Comment:
[Consider][Minor] I know you addressed this comment already, at at the risk of bikeshedding, this still comes off as though we had a direct need to force caching, rather than it merely being an (unfortunate) side effect.

```suggestion
    // Deliberately use a valid fixed timestamp, which will force caching
    // until we implement a more considered method.
    // TODO(DF): Consider how best to handle this via OpenAssetIO.
```